File: Backend/Models/AppointmentAnimalBooking.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace Models
{
    public class AppointmentAnimalBooking : Common
    {
        [Required(ErrorMessage = "Der skal være mindst en tilkoblet aftale")]
        public required int AppointmentId { get; set; }
        public Appointment? Appointment { get; set; }

        [Required(ErrorMessage = "Der skal være mindst et tilkoblet dyr")]
        public required int AnimalId { get; set; }
        public Animal? Animal { get; set; }
    }
}

File: flutter_app/lib/classes/objects/appointment_animal_booking.dart
import 'animal.dart';
import 'appointment.dart';
import 'common.dart';

class AppointmentAnimalBooking extends Common {
  AppointmentAnimalBooking({
    required this.id,
    required this.AppointmentId,
    this.appointment,
    required this.AnimalId,
    this.animal,
    this.createdAt,
    this.updatedAt,
  });
  final String AppointmentId;
  final Appointment? appointment;

  final String AnimalId;
  final Animal? animal;

  @override
  final String id;

  @override
  final DateTime? createdAt;

  @override
  final DateTime? updatedAt;

  factory AppointmentAnimalBooking.fromJson(Map<String, dynamic> json) {
    return AppointmentAnimalBooking(
      id: json['id'] as String,
      AppointmentId: json['appointmentId'] as String,
      appointment: json['appointment'] == null
          ? null
          : Appointment.fromJson(json['appointment']),
      AnimalId: json['animalId'] as String,
      animal: json['animal'] == null ? null : Animal.fromJson(json['animal']),
      createdAt: DateTime.tryParse(json['createdAt']),
      updatedAt: DateTime.tryParse(json['updatedAt']),
    );
  }
}
